### PR TITLE
Multi stage build

### PR DESCRIPTION
**Pull request by Nick-Triller**
Hi, thank you for this useful project.

I introduced a multi-stage docker build as I did not want to install the go toolchain locally. 
The deployer binary is built in a docker container and transferred to a different container afterwards. 
The resulting image is as small as before (same base image without go tools).

This change might be useful for continuous integration as a build agent would only need docker. 
Feel free to reject this PR if you prefer your current setup.

Thanks again!

### CODE REVIEW

**Comment by Nick-Triller** [comment]
For some reason `docker login` failed in the travis build. I am not sure why.

**Comment by nielsdenissen** [comment]
Hi Nick, looks like a nice addition, I'll have a look at it!
Regarding your `docker login` error, we've setup travis secrets to login to Dockerhub when pushing the docker images. These secrets are only valid for the `ing-bank/flink-deployer` repo. Could be that's the reason it doesn't work on your fork. Perhaps it's better we start building docker images only on master

**Comment by mrooding** [comment]
#21 which is just merged to master disabled builds for anything other than master/tags. Can you rebase your branch on the latest master incorporate these changes? 🙏

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,11 @@
-FROM alpine:3.8
-
-COPY cli /flink-deployer/cli
+FROM golang:1.10.4-alpine3.8
```
[the remark this answers is not in the document]
Comment (Nick-Triller) [reply Dockerfile]: Good point!

**Comment by Nick-Triller** [comment]
> 
> 
> #21 which is just merged to master disabled builds for anything other than master/tags. Can you rebase your branch on the latest master incorporate these changes? 🙏

Sure, I rebased onto master.

**Pull request merged**